x-pack/plugins/infra/server/lib/alerting/log_threshold/log_threshold_executor.ts

```diff
@@ -101,17 +169,124 @@ const getESQuery = (
   // Negative assertions (things that "must not" match)
   const mustNotFilters = buildFiltersForCriteria(negativeCriteria);
 
-  const query = {
+  const rangeFilter = {
+    range: {
+      [timestampField]: {
+        gte: from,
+        lte: to,
+        format: 'epoch_millis',
+      },
+    },
+  };
+
+  return { rangeFilter, mustFilters, mustNotFilters };
+};
+
+const getGroupedESQuery = (
+  params: LogDocumentCountAlertParams,
+  sourceConfiguration: InfraSource['configuration'],
+  index: string
+): object | undefined => {
+  const { groupBy } = params;
+
+  if (!groupBy || !groupBy.length) {
+    return;
+  }
+
+  const timestampField = sourceConfiguration.fields.timestamp;
+
+  const { rangeFilter, mustFilters, mustNotFilters } = buildFiltersFromCriteria(
+    params,
+    timestampField
+  );
+
+  // For group by scenarios we'll pad the interval by 20% on the left (lte) and right (gte), this is so
+  // a wider net is cast to "capture" the groups. This is to account for scenarios where we want ascertain if
+  // there were "no documents" (less than 1 for example). In these cases we may be missing documents to build the groups
+  // and match (or not match) the criteria.
```

Comment:
In my tests that mostly lead to the disappearance of groups triggering an alert, good job :+1: I wonder if padding by one whole interval size might be more suited for the scenario we're looking for. That way it would be less dependent on the distribution of the documents within the interval buckets. What would be the risk?

Reply:
I don't think there's much risk pushing up to 1 x interval on each side. I think that should be the maximum we push up to before behaviour "deviates". But I'll make the change to 1x 👍